documentation/docs/guides/creating-plans.md

```diff
@@ -0,0 +1,305 @@
+( O)> please provde a generic plan
+```
+While Goose creates a standardized plan format, it customizes the content based on your answers. Goose can generate the code needed to implement the steps of the plan it produces. You should review the plan and any code that it generates before ending plan mode (`/endplan`) and asking Goose to implement the plan. 
+
+Below is a sample plan for this project, with the generated website code omitted for brevity:
+
+```bash
+# Plan for Converting CLI Script to React Web Application
+
+## Step 1: Set up the React project and dependencies
+1. Create a new React application using Create React App
+2. Install necessary dependencies
+3. Set up project structure
+   src/
+   ├── components/
+   │   ├── Auth/
+   │   ├── Layout/
+   │   ├── Search/
+   │   ├── Results/
+   │   └── History/
+   ├── services/
+   │   ├── contentful.ts
+   │   ├── auth.ts
+   │   └── storage.ts
+   ├── utils/
+   │   ├── validation.ts
+   │   └── helpers.ts
+   ├── contexts/
+   │   └── AppContext.tsx
+   └── pages/
+       ├── Home.tsx
+       ├── Search.tsx
+       └── History.tsx
+
+## Step 2: Implement authentication with Okta
+1. Create an Okta application in the Okta Developer Console
+2. Configure the Okta authentication settings in the application
+3. Implement protected routes using Okta's React components
+## Step 3: Create the layout and UI components with Material UI
+1. Implement the main layout with Material UI
+2. Create the search form component
+3. Create the results display component with three columns
+4. Implement the "Load More" pagination component
+## Step 4: Implement Contentful service
+1. Create a service for interacting with Contentful
+## Step 5: Implement local storage service for history
+1. Create a service for managing search history in localStorage
+## Step 6: Implement validation utilities
+1. Create utility functions for input validation
+## Step 7: Implement main pages
+1. Create the Search page
+2. Create the History page
+## Step 8: Set up routing and main application
+1. Create the main App component
+## Step 9: Implement error handling and loading states
+1. Create error boundary components
+2. Add loading indicators for API operations
+3. Implement error messages display
+
+## Step 10: Set up deployment configuration
+1. Create AWS deployment configuration
+## Step 11: Testing and quality assurance
+1. Write unit tests for key components
+2. Implement integration tests for the main workflows
+3. Perform manual testing of the application
+
+This plan provides a comprehensive framework for converting your CLI script to a React web application with Material UI and Okta authentication. You'll need to adapt specific parts based on your exact requirements and the functionality of your original script.
+
+```
+
+## A simple construction plan example
+Goose can produce good plans for relatively simple projects such as the home construction example:
```

Comment:
yeah maybe we can bring this example up as the first example..and then put the more complex examples near the bottom..so we can incrementally show /plan capabilities